plugins/modules/identity/keycloak/keycloak_user.py

```diff
@@ -0,0 +1,530 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: keycloak_user
+
+short_description: Allows administration of Keycloak users via Keycloak API
+
+description:
+    - This module allows you to add, remove or modify Keycloak users via the Keycloak REST API.
+      It requires access to the REST API via OpenID Connect; the user connecting and the client being
+      used must have the requisite access rights. In a default Keycloak installation, admin-cli
+      and an admin user would work, as would a separate client definition with the scope tailored
+      to your needs and a user having the expected roles.
+
+    - The names of module options are snake_cased versions of the camelCase ones found in the
+      Keycloak API and its documentation at U(https://www.keycloak.org/docs-api/8.0/rest-api/index.html).
+
+    - Attributes are multi-valued in the Keycloak API. All attributes are lists of individual values and will
+      be returned that way by this module. You may pass single values for attributes when calling the module,
+      and this will be translated into a list suitable for the API.
+
+    - When updating a user, where possible provide the user ID to the module. This removes a lookup
+      to the API to translate the username into the user ID.
+
+
+options:
+    state:
+        description:
+            - State of the user.
+            - On C(present), the user will be created if it does not yet exist, or updated with the parameters you provide.
+            - On C(absent), the user will be removed if it exists.
+        default: 'present'
+        type: str
+        choices:
+            - present
+            - absent
+
+    username:
+        type: str
+        description:
+            - Username of the user.
+            - This parameter is required only when creating or updating the user.
+
+    realm:
+        type: str
+        description:
+            - They Keycloak realm under which this user resides.
+        default: 'master'
+
+    access:
+        type: dict
+        description:
+            - The access.
+    attributes:
+        type: dict
+        description:
+            - A dict of key/value pairs to set as custom attributes for the user.
+            - Values may be single values (e.g. a string) or a list of strings.
```

Comment:
```suggestion
            - Values may be single values (for example a string) or a list of strings.
```
Can they also be other types than strings, say integers or floats?